src/Tools/ExternalAccess/Razor/RazorDynamicFileInfoProviderWrapper.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Razor
+{
+    [Shared]
+    [ExportMetadata("Extensions", new string[] { "cshtml", "razor", })]
+    [Export(typeof(IDynamicFileInfoProvider))]
+    internal sealed class RazorDynamicFileInfoProviderWrapper : IDynamicFileInfoProvider
+    {
+        private readonly IRazorDynamicFileInfoProvider _innerDynamiFileInfoProvider;
+
+        public event EventHandler<string> Updated;
+
+        [ImportingConstructor]
+        public RazorDynamicFileInfoProviderWrapper(
+            [Import(AllowDefault = true)] IRazorDynamicFileInfoProvider innerDynamiFileInfoProvider)
```

Comment:
>innerDynamiFileInfoProvider [](start = 72, length = 27)

typo - missing `c`.

Reply:
We decided that we'll keep using MEF component rejection, so this doesn't need to be `AllowDefault = true`. It should however be `Lazy<IRazorDynamicFileInfoProvider>`, so that we don't load Razor in projects that are not Razor projects.

---
In reply to: [392333118](https://github.com/dotnet/roslyn/pull/42174#discussion_r392333118) [](ancestors = 392333118)